lib/nuget-parser/dotnet-core-parser.js

```diff
@@ -123,38 +121,41 @@ function getTargetObjToRun(manifest) {
   return manifest.targets[selectedTargetKey];
 }
 
+module.exports = {
+  parse: function (tree, fileContent, fileContentParser) {
+    return new Promise(function parseFileContents(resolve) {
+      debug('Trying to parse dot-net-cli manifest');
+      const manifest = fileContentParser.parse(fileContent);
 
-function parse(fileContent, packageTree) {
-  const manifest = JSON.parse(fileContent);
-
-  if (!manifest.project) {
-    throw new Error('Project field was not found in project.assets.json');
-  }
-
-  if (manifest.project.version) {
-    packageTree.version = manifest.project.version;
-  }
+      if (!manifest.project) {
+        throw new Error('Project field was not found in project.assets.json');
+      }
 
-  const selectedFrameworkObj = getFrameworkObjToRun(manifest);
-  const selectedTargetObj = getTargetObjToRun(manifest);
+      if (manifest.project.version) {
+        tree.version = manifest.project.version;
+      }
 
-  initFreqDepsDict();
+      tree.meta.targetFramework = getFrameworkToRun(manifest);
+      const selectedFrameworkObj = manifest.project.frameworks[tree.meta.targetFramework];
+      const selectedTargetObj = getTargetObjToRun(manifest);
```

Comment:
This way you are breaking backward compatibility with old version.

First of all you should use TF from proj file (as it was before). And only if TF not defined in proj file - use TF from project.assets.json.

Am I right?

Reply:
you are correct. Fixing